examples/bri-3/src/bri/auth/api/auth.controller.ts

```diff
@@ -1,14 +1,21 @@
-import { Controller, Post, Request } from '@nestjs/common';
+import { Controller, Get, Post, Request } from '@nestjs/common';
 import { Public } from 'src/bri/decorators/public-endpoint';
-import { AuthService } from '../auth.service';
+import { AuthService } from '../services/auth.service';
+import { generateMnemonic } from 'bip39';
 
 @Controller('auth')
 export class AuthController {
-  constructor(private authService: AuthService) {}
+  constructor(private readonly authService: AuthService) {}
 
   @Public()
   @Post('login')
   async login(@Request() req) {
     return this.authService.login(req.body);
   }
+
+  @Public()
+  @Get('mnemonic')
+  async mnemonic(@Request() req) {
+    return generateMnemonic();
```

Comment:
If someone knows a message and a signature of that message for a specific public key, they can pass the login as that public key. This is because the mnemonic (one time pass)  is not stored during geneation so we do not check if it was issued by our backend during login. Do we really want this?

@Manik-Jain @skosito @Therecanbeonlyone1969 @Kasshern @biscuitdey @Ybittan

Reply:
yea... we need to store some nonce that is signed and then when authenticating pull the last nonce from db, not send it in request body
also, why using bip39 mnemonic at first place as thing to be signed? lets just store some incrementing nonce or uuid or something like that...

Reply:
@ognjenkurtic Right, I missed on the OTP storage part. Will push in the code soon.
@skosito the incrementing nonce approach works, and I think this is legacy. bip39 mnemonics can act as the random string of words to be signed, and updated in DB  the hash for each new generated mnemonic, similar to a nonce.

Another approach to it can be, we store in the signature instead of the nonce/mnemonic, to identify if a jwt token was linked to the provided signature. This way, we cover both the scenario

Reply:
I agree a uuid is sufficient as a secret to be signed. The secret should be encrypted to the public key of the requesting user such that decryption requires the corresponding private key.

For replay protection, we need an incrementing nonce and the specific BPI domain to be accessed to be returned in the query response with the JWT, ideally as part of the JWT.

Reply:
latest commit has the nonce storage changes